third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -843,8 +890,11 @@ LogicalResult PointerCanonicalizer::rewriteFunction(triton::FuncOp funcOp) {
       return failure();
 
     // Clean-up
-    for (Operation *op : llvm::reverse(opToDelete))
-      op->erase();
+    for (Operation *op : opToDelete) {
+      op->dropAllReferences();
```

Comment:
Shouldn't this be an assert(op->use_empty()) at this point? Seems dangerous to just drop references. Or is this caused by opToDelete order dependencies?

Reply:
Ah, I see your comment above answers this.

Reply:
I don't see comments in the above? still good to add some comments here. :)

Reply:
I think Simon meant what I wrote in the PR description. I added the comment in the code as well :)